backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -74,4 +78,23 @@ public FolderResponse getFolderDetails(UUID uuid) {
     FolderResponse folderResponse = modelMapper.map(folderEntity, FolderResponse.class);
     return folderResponse;
   }
+
+  @Override
+  public List<FolderResponse> getAllRootFolders(UUID bucketId) {
+    List<FolderEntity> allFolders =
+        folderRepository.findAllByBucketIdAndParentFolderIsNull(bucketId);
+    return getFolders(allFolders);
+  }
+
+  @Override
+  public List<FolderResponse> getAllFoldersWithParent(UUID parentFolderId) {
+    List<FolderEntity> allFolders = folderRepository.findAllByParentFolderId(parentFolderId);
+    return getFolders(allFolders);
+  }
+
+  private List<FolderResponse> getFolders(List<FolderEntity> allFolders) {
```

Comment:
I think it makes more sense for this to be named toResponse or something like that, as the function just converts a list of entities to a list of responses.

Reply:
Maybe toResponses or toResponseList, I agree the name maybe can decieve at first glance..